### PR TITLE
Remove build tag `unit` and move `MockActors` & `FailingActors` to a separate testing package under `pkg/actors` 

### PR DESCRIPTION
**Pull request by dazhao-msft**
# Description

The change is to get rid of build tag `unit` and instead move `MockActors` and `FailingActors` to a separate testing package under `pkg/actors`. This also follows the same convention by other implementations of mocks.

## Issue reference

Please reference the issue this PR will close: #5685

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Unit tests passing
* [x] End-to-end tests passing

### CODE REVIEW

**Review by mukundansundar — DISMISSED** [review]
lgtm.

Thanks for the contribution!

**Review by ItalyPaleAle — DISMISSED** [review]

**Review by berndverst — DISMISSED** [review]

**Comment by dazhao-msft** [comment]
@mukundansundar @ItalyPaleAle @berndverst thanks guys for reviewing the change. Please also help merge the PR (or let me know if any additional steps are required before merge).

**Comment by berndverst** [comment]
@dazhao-msft we cannot merge as we are not maintainers of this repo. All maintainers are out of the office (at their respective businesses) until next week.

**Review by mukundansundar — APPROVED** [review]

**Comment by dazhao-msft** [comment]
@ItalyPaleAle I'm removing `buildinfo_unit.go` which was added via PR #5677, as the file introduced a new occurrence of the `unit` tag, which is being removed by this PR. I think it makes more sense to remove `buildinfo_unit.go` because unit tests shouldn't depend on feature flags to run. Instead, it should be the integration tests that verify the impacts of feature flags. In addition, all the current unit tests can pass without `buildinfo_unit.go`.

BTW, what else needs to be done to merge this PR? I notice that it has been marked as autoupdate for a while.

**Comment by ItalyPaleAle** [comment]
> In addition, all the current unit tests can pass without buildinfo_unit.go.

Yes, that's correct because Resiliency is now enabled everywhere.

However, we will need a way to hardcode feature flags for unit tests I'm afraid.

**Comment by dazhao-msft** [comment]
> However, we will need a way to hardcode feature flags for unit tests I'm afraid.

Not really. Individual features should be unit tested independently. If a component depends on another with feature flag, ideally the dependency should be mocked up rather than turning on/off the actual feature flag with the code like `buildinfo_unit.go`.

**Comment by ItalyPaleAle** [comment]
The problem is during “migrations”, before we enable a feature flag. That’s when I added the Resiliency one, a few days before enabling it everywhere. In this case, the goal was to test that _all_ unit tests would work even with Resiliency enabled.

**Comment by dazhao-msft** [comment]
> The problem is during “migrations”, before we enable a feature flag. That’s when I added the Resiliency one, a few days before enabling it everywhere. In this case, the goal was to test that _all_ unit tests would work even with Resiliency enabled.

Actually would it make sense to also append `$(GCFLAGS) -ldflags=$(LDFLAGS)` to `go test` and `gotestsum` to keep consistent with `go build`?

File: pkg/buildinfo/buildinfo_unit.go
Comment (ItalyPaleAle) [inline pkg/buildinfo/buildinfo_unit.go:24]: Sorry, we really need this file to be able to hardcode certain features :(

Comment (dazhao-msft) [reply pkg/buildinfo/buildinfo_unit.go:24]: How about adding the same arguments of `ldflags` to `go test`?

I made the changes in the PR a few days ago in the Makefile.

Comment (dazhao-msft) [reply pkg/buildinfo/buildinfo_unit.go:24]: In general, I think it's better to keep the arguments of `go build` and `go test` as consistent as possible, so it tests the same behaviors as it ships.

**Pull request closed without merging**
